content/docs/static-type-checking.md

```diff
@@ -0,0 +1,65 @@
+---
+id: static-type-checking
+title: Static Type Checking
+permalink: docs/static-type-checking.html
+prev: typechecking-with-prototypes.html
+next: refs-and-the-dom.html
+---
+
+As your app grows, you can catch and eliminate an entire category of bugs with static typechecking. You can use JavaScript extensions like [Flow](https://flowtype.org/) or [TypeScript](https://www.typescriptlang.org/) to statically typecheck your whole application. In complex applications, we recommend to use Flow or TypeScript for typechecking instead of typechecking with PropTypes.
+
+## Using Flow with React
+
+For trying out Flow, use this [playground](https://flow.org/try/).
+
+Flow and [Babel](http://babeljs.io/) work well together, so it doesn’t take much to adopt Flow as a React user who already uses Babel.
+
+### With babel
+
+Flow and Babel are designed to work great together. It takes just a few steps to set them up together.
+
+If you don’t have Babel setup already, you can do that by following [this guide](http://babeljs.io/docs/setup/).
+
+Once you have Babel setup, install `babel-preset-flow` with either [Yarn](https://yarnpkg.com/) or [npm](https://www.npmjs.com/).
+
+```bash
+yarn add --dev babel-preset-flow
+# or
+npm install --save-dev bebel-preset-flow
+```
+
+Then add `flow` to your Babel presets config.
+
+```
+{
+  "presets": ["flow"]
+}
+```
```

Comment:
This section only explains how to add Flow syntax support to a project using Babel. But it doesn't actually explain how to run Flow.